queries.sql

```diff
@@ -9,3 +9,50 @@ SELECT * from animals WHERE neutered = true;
 SELECT * from animals WHERE name != 'Gabumon';
 SELECT * from animals WHERE weight_kg <= 17.3 AND weight_kg >= 10.4;
 
+/* transactions */
+BEGIN;
+UPDATE animals
+SET species = 'unspecified'
+WHERE species IS null;
+COMMIT;
+
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE neme LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS null;
+COMMIT;
+
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+
+BEGIN;
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT delete_2022_animals;
+
+UPDATE animals
+SET weight_kg = weight_kg * (-1);
+
+ROLLBACK TO delete_2022_animals;
+
+UPDATE animals
+SET weight_kg = weight_kg * (-1)
+WHERE weight_kg < 0;
+COMMIT;
+
+
+SELECT COUNT(id) from animals;
+SELECT COUNT(id) from animals WHERE escape_attempts = 0 ;
+SELECT AVG(weight_kg) as avg_weight FROM animals;
+SELECT SUM(escape_attempts) AS escape_attempts, neutered FROM animals GROUP BY neutered;
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight from animals GROUP BY species;
+SELECT species, AVG(escape_attempts) from animals WHERE date_of_birth between '1990-01-01' AND '2000-01-01'
```

Comment:
- [ ] According to this query: ` What is the average number of escape attempts per animal type of those born between 1990 and 2000?` You must include all the days and months of the year 2000 instead of only the first day of January. The response might look like this 👇🏻 

   ```sql
      SELECT species, AVG(escape_attempts) FROM animals
      GROUP BY species
      WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31';
   
   ```